File: python/examples/scoreLimits.py
from aristech_nlp_client import NlpClient, GetScoreLimitsRequest
from utils import host, auth_token, auth_secret, root_cert, ssl, project_id

client = NlpClient(host=host, ssl=ssl, root_cert=root_cert, auth_token=auth_token, auth_secret=auth_secret)
response = client.score_limits(GetScoreLimitsRequest(
  project_id=project_id,
  test_sentences_upper_limit=[
    "What are the lottery numbers?",
    "Can you tell me the winning numbers?",
  ],
  test_sentences_lower_limit=[
    "What's the weather like?",
    "I want to speak to a human.",
],
))
print(response)

File: rust/examples/scoreLimits.rs
mod utils;
use std::error::Error;
use utils::get_tls_options;

use aristech_nlp_client::{get_client, get_score_limits, nlp_service::GetScoreLimitsRequest};

#[tokio::main]
async fn main() -> Result<(), Box<dyn Error>> {
    // Load environment variables from .env file
    dotenv::dotenv().ok();

    let host = std::env::var("HOST")?;
    let tls_options = get_tls_options()?;
    let mut client = get_client(host, tls_options).await?;

    let response = get_score_limits(
        &mut client,
        GetScoreLimitsRequest {
            project_id: std::env::var("PROJECT_ID")?,
            test_sentences_upper_limit: vec![
                "What are the lottery numbers?".to_string(),
                "Can you tell me the winning numbers?".to_string(),
            ],
            test_sentences_lower_limit: vec![
                "What's the weather like?".to_string(),
                "I want to speak to a human.".to_string(),
            ],
        },
    )
    .await?;
    println!("{:?}", response);
    Ok(())
}
